src/java.base/share/native/libzip/zip_util.c

```diff
@@ -863,8 +868,16 @@ ZIP_Put_In_Cache0(const char *name, ZFILE zfd, char **pmsg, jlong lastModified,
     zip->lastModified = lastModified;
 
     if (zfd == -1) {
-        if (pmsg && getLastErrorString(errbuf, sizeof(errbuf)) > 0)
+#ifdef _WIN32
```

Comment:
1. Could we use strerror_s here?

2. It's better to encapsulate similar code to a separate function or macro, rather than add #ifdef _WIN32 in the middle of the code.